### ISSUE TITLE
Is it necessary to write comments in native language?

### ISSUE THREAD

**Issue by Unknown-Chinese-User** [trekhleb/state-of-the-art-shitcode#11]
I know it's a ironical project, but I have question whether to write comments in native language.
Using Chinese will has trouble shifting input method, and it would lead to many problems between win and linux.
But using English will avoid these questions as well as practice Computer English, and it wouldn't cause any problems reading it.

**Comment by stepfencurryxiao** [comment]
I agree with @Unknown-Chinese-User that it is better to write English comments than Chinese comments.

**Comment by CSharperMantle** [comment]
Well some Chinese coders will name their vars and write their comments in **Pinyin**, like `YongHuShuJuDao` rather than `UserDataDao`. This should be considered as a culture-specific guideline of shitcode. XD

**Comment by 66Leo66** [comment]
@CSharperMantle Talking about **Pinyin**, I'm gonna say that I prefer reading **English** comments than **Pinyin**. Pinyin in codes is totally just SHIT💩💩💩

**Comment by z-t-y** [comment]
`
  let shuju       =“henzhongyaodeshuju”
           let yifangwanyi = shuju;
 var zaiciyifangwanyi = yifangwanyi
         console.log(shuju)
// zhe xie dou shi shi💩💩💩
`

**Comment by z-t-y** [comment]
`let shuju =“henzhongyaodeshuju”; let yifangwanyi = shuju; var zaiciyifangwanyi =yifangwanyi; console.log(shuju); // zhe xie dou shi shi💩💩💩`

**Comment by z-t-y** [comment]
This is Chinese **Pinyin**!

**Comment by CSharperMantle** [comment]
@z-t-y This is absolutely shitcode and exactly what I mean! :hankey:

**Comment by KirkHadley** [comment]
Seems pretty reasonable to write code in the language of your intended users.... If you don't speak English, you write shit code? That doesn't seem quite right. I get that it's just a silly little repo, but when one has an audience, perhaps one should aim for a higher level of conscientiousness. And also maybe you're just wrong- perhaps Pinyin comments are why Didi so thoroughly smoked yall 🤣

**Comment by Miigon** [comment]
I speak Chinese as a native language and for me codes in pinyin looks like it's generated by a random character generator...

**Comment by CodeCrazy-ywt** [comment]
Or, we can even write codes using Chinese Characters!!!😄😄😄 For example:
```python
def 输出(内容):
    print(内容)

数据='你好世界'
输出(数据)
```

**Comment by MrHeavythree** [comment]
。。。。。。人才

**Comment by CodeCrazy-ywt** [comment]
> @CSharperMantle Talking about **Pinyin**, I'm gonna say that I prefer reading **English** comments than **Pinyin**. Pinyin in codes is totally just SHIT💩💩💩

Yeah, you must read it out to get the meaning. (To make things worse, you need to guess the tone!)
确实，你先要读一遍才能知道是什么意思（而且你得先知道音调！）
